web/server/www/scripts/codecheckerviewer/ListOfRuns.js

```diff
@@ -92,7 +92,24 @@ function (declare, dom, ObjectStore, Store, Deferred, topic, Dialog, Button,
     },
 
     getIdentity : function (run) {
-      return run.runId;
+      return run.runid;
+    },
+
+    get : function (id) {
+      var deferred = new Deferred();
+
+      var runFilter = new CC_OBJECTS.RunFilter();
+      runFilter.ids = [ id ];
+
+      CC_SERVICE.getRunData(runFilter, 1, 0, function (runDataList) {
+        if (typeof runDataList === 'string') {
+          deferred.reject('Failed to get run ' + id + ': ' + runDataList);
+        } else {
+          deferred.resolve(runDataList[0]);
+        }
+      }).fail(function (xhr) { util.handleAjaxFailure(xhr); });
```

Comment:
Shouldn't we call the deferred.reject in the fail section too, and provide some additional error message to the user that the run removal failed? What if the delete query times out?

Reply:
There is a ticket for fixing this at every occurrence: #2257.